src/search_algo.hpp

```diff
@@ -542,7 +657,99 @@ search(LocalDataHolder<TGlobalHolder> & lH)
 
                     ++lH.stats.hitsAfterSeeding;
 
-                    if (!seedLooksPromising(lH, m))
+                    if (!seedLooksPromisingFwd(lH, m))
+                        ++lH.stats.hitsFailedPreExtendTest;
+                    else
+                        lH.matches.push_back(m);
+                }
+            }
+        }
+    }
+}
+
+template <typename TGlobalHolder>
+inline void
+search_rev(LocalDataHolder<TGlobalHolder> & lH)
```

Comment:
Please implement this via an extra template argument to avoid copy'n'pasting the entire function!